test/common/stats/stat_test_utility.cc

```diff
@@ -0,0 +1,119 @@
+#include "test/common/stats/stat_test_utility.h"
+
+#include "common/memory/stats.h"
+
+namespace Envoy {
+namespace Stats {
+namespace TestUtil {
+
+bool hasDeterministicMallocStats() {
+  // We can only test absolute memory usage if the malloc library is a known
+  // quantity. This decision is centralized here. As the preferred malloc
+  // library for Envoy is TCMALLOC that's what we test for here. If we switch
+  // to a different malloc library than we'd have to re-evaluate all the
+  // thresholds in the tests referencing hasDeterministicMallocStats().
+#ifdef TCMALLOC
+  const size_t start_mem = Memory::Stats::totalCurrentlyAllocated();
+  std::unique_ptr<char[]> data(new char[10000]);
+  const size_t end_mem = Memory::Stats::totalCurrentlyAllocated();
+  return end_mem - start_mem >= 10000; // actually 10240
+#else
+  return false;
+#endif
+}
+
+void forEachStat(int num_clusters, std::function<void(absl::string_view)> fn) {
+  // These are stats that are repeated for each cluster as of Oct 2018.
+  static const char* cluster_stats[] = {"bind_errors",
```

Comment:
I strongly suspect this could be auto generated using the same macro magic we use to define all the stats in prod code? Or if that doesn't work creating a cluster stat scope and seeing what stats are in it? Is it possible to look into that so we avoid this getting out of date?